File: GotravelDesktop/src/main/java/com/gotravel/Model/Etapa.java
package com.gotravel.Model;

import com.gotravel.Utils.Fechas;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Etapa {

    private int id;
    private String nombre;
    private String fechaInicio;
    private String fechaFinal;
    private Double costeTotal;
    private String tipo;
    private List<Servicio> contrataciones;

    public Etapa(String nombre, String fechaInicio, String fechaFinal, String tipo, Double costeTotal) {
        this.nombre = nombre;
        this.fechaInicio = fechaInicio;
        this.fechaFinal = fechaFinal;
        this.tipo = tipo;
        this.costeTotal = costeTotal;
    }

    public String inicio() {
        return LocalDate.parse(this.fechaInicio, Fechas.formatoFromDb).format(Fechas.formatoFinal);
    }

    public String fin() {
        return LocalDate.parse(this.fechaFinal, Fechas.formatoFromDb).format(Fechas.formatoFinal);
    }

}

File: GoTravelMobile/app/src/main/java/com/gotravel/mobile/data/model/Etapa.kt
package com.gotravel.mobile.data.model

import android.os.Build
import androidx.annotation.RequiresApi
import com.gotravel.mobile.ui.utils.formatoFinal
import com.gotravel.mobile.ui.utils.formatoFromDb
import kotlinx.serialization.Serializable
import lombok.AllArgsConstructor
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.Setter
import java.time.LocalDate

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Serializable
data class Etapa(
    val id: Int? = null,
    val nombre: String,
    val fechaInicio: String,
    val fechaFinal: String,
    val costeTotal: Double,
    val tipo: String
) {
    val contrataciones: List<Contratacion>? = null

    val inicio: String
        @RequiresApi(Build.VERSION_CODES.O)
        get() {
            return LocalDate.parse(fechaInicio, formatoFromDb).format(formatoFinal)
        }

    val final: String
        @RequiresApi(Build.VERSION_CODES.O)
        get() {
            return LocalDate.parse(fechaFinal, formatoFromDb).format(formatoFinal)
        }

}